src/sync/pull-request.ts

```diff
@@ -70,42 +60,35 @@ export const getPullRequestTask = async (
 		// Fetch in parallel instead. Given that's an expermient for a single customer, let's not
 		// overcomplicate it too much and limit ourselves to 2 pages.
 		const limitedPageSizeCoef = Math.min(5, pageSizeCoef);
-		const shouldFetchNextPage = pageSizeCoef > 5;
-
-		const prFetchJobs: Promise<any>[] = [];
+		const shouldFetchNextPageInParallel = pageSizeCoef > 5;
 
 		const scaledPageSize = perPage * limitedPageSizeCoef;
-		const scaledCursor = Math.max(1, Math.floor(Number(cursor) / limitedPageSizeCoef));
 
-		prFetchJobs.push(doGetPullRequestTask(
-			logger, gitHubInstallationClient, jiraHost, repository,
-			scaledCursor,
-			scaledPageSize
-		));
-
-		if (shouldFetchNextPage) {
-			logger.info("Fetch second page in parallel");
-			prFetchJobs.push(doGetPullRequestTask(
-				logger, gitHubInstallationClient, jiraHost, repository,
-				scaledCursor + 1,
-				scaledPageSize
-			));
-		}
-
-		const [page1, page2] = await Promise.all(prFetchJobs);
-
-		const mergeResult = {
-			edges: [...(page1?.edges || []), ...(page2?.edges || [])],
-			jiraPayload:
-				mergeJiraPayload(page1?.jiraPayload, page2?.jiraPayload)
-		};
+		// Cursor 1, 2, 3, 4, 5 should be mapped to scaled cursor 1;
+		// Cursor 6, 7, 8, 9, 10 shoul be mapped to scaled cursor 2;
+		// etc
+		// Given that the page counter starts from 1, we need to deduct 1 first and then add 1 back to the outcome
+		const scaledCursor = 1 + Math.floor((Number(cursor) - 1) / limitedPageSizeCoef);
 
-		const nextPageCursor = Number(cursor) + (shouldFetchNextPage ? limitedPageSizeCoef * 2 : limitedPageSizeCoef);
-
-		mergeResult.edges.forEach((edge) => {
-			edge.cursor = nextPageCursor;
+		const data = await fetchNextPagesInParallel(
+			shouldFetchNextPageInParallel ? 2 : 1,
+			scaledCursor,
+			(scaledPageNoToFetch) =>
+				doGetPullRequestTask(
+					logger, gitHubInstallationClient, jiraHost, repository,
+					scaledPageNoToFetch,
+					scaledPageSize
+				)
+		);
+		(data.edges || []).forEach(edge => {
```

Comment:
Are we saying we want to loop over an empty array if data.edges doesn't exist is empty? Think we need to add a test to see what happens if edges doesn't exist on data/edges is empty.

Reply:
We already have such test: http://github.com/atlassian/github-for-jira/blob/b68a8348d0a73ac2a71c5c99bd59a6437cdcff0d/src/sync/pull-requests.test.ts#L408-L408

Reply:
Good stuff.